flink-core/src/main/java/org/apache/flink/types/RowKind.java

```diff
@@ -47,10 +47,69 @@
 	 * needs to retract the previous row first. OR it describes an idempotent update, i.e., an update
 	 * of a row that is uniquely identifiable by a key.
 	 */
-	UPDATE_AFTER,
+	UPDATE_AFTER("UA", (byte) 2),
 
 	/**
 	 * Deletion operation.
 	 */
-	DELETE
+	DELETE("D", (byte) 3);
+
+	private final String shortString;
+
+	private final byte value;
+
+	/**
+	 * Creates a {@link RowKind} enum with the given short string and byte value representation of
+	 * the {@link RowKind}.
+	 */
+	RowKind(String shortString, byte value) {
+		this.shortString = shortString;
+		this.value = value;
+	}
+
+	/**
+	 * Returns a short string representation of this {@link RowKind}.
+	 *
+	 * <p><ul>
```

Comment:
I would remove this list. It is redundant and defined in the enum value already. Same for the list below.

Reply:
That's true. But if users are looking at the Javadoc website, the comment is the only thing he/she can see, not the code. That's why I put the list in the Javadoc.

Reply:
But users that only look at JavaDocs are usually not the ones that should work with byte values for performance reasons :D

Reply:
Yes. But end users may care about the short string representation?